pennylane/_queuing.py

```diff
@@ -236,8 +238,11 @@ def __init__(self):
     def __exit__(self, exception_type, exception_value, traceback):
         super().__exit__(exception_type, exception_value, traceback)
 
-        # Remove duplicates that might have arisen from measurements
-        self.queue = list(OrderedDict.fromkeys(self.queue))
+        # Remove duplicates that might have arisen from measurements.
+        # Code taken from https://stackoverflow.com/questions/480214
+        seen = set()
+        seen_add = seen.add
+        self.queue = [x for x in self.queue if not (x in seen or seen_add(x))]
```

Comment:
Out of curiosity, is the `seen_add = seen.add` a micro-optimization, to avoid the method lookup during the iteration?

Reply:
Yes, I stole it from https://stackoverflow.com/questions/480214/how-do-you-remove-duplicates-from-a-list-whilst-preserving-order

It's probably totally immeasurable speedup, so I'll just revert it.

Reply:
> It's probably totally immeasurable speedup, so I'll just revert it.

It might not be! I once saved 0.15s simply by merging two methods and removing the method call between them 🤦